coralproject/talk
# DSA Notifications

## What does this PR do?

Preliminarily adds notifications to the DSA epic branch.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

- Adds a new `Notification` type and corresponding Connection types

## Does this PR introduce any new environment variables or feature flags?

No.

## If any indexes were added, were they added to `INDEXES.md`?

```
db.notifications.createIndex({ tenantID: 1, id: 1 }, { unique: true });
```

This index creates the uniqueness constraint for the `tenantID` and `id` fields on the notifications collection

```
db.notifications.createIndex({ tenantID: 1, ownerID: 1, createdAt: 1 });
```

This index speeds up the retrieval of notifications by `tenantID`, `ownerID`, and `createdAt` which is the most common way of retrieving notifications for pagination in the notifications tab on the stream.

## How do I test this PR?

- Enable the DSA features under `Admin > Configure > General`
- Create some comments
- Feature, approve, or reject the comments
- Check `Notifications` for updates on these actions

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

- Merge into the dsa-launchpad branch
- Eventually merge the dsa-launchpad branch into develop
- Release
- Ensure appropriate indices (described above) are applied to any mongo instances using the notification features
